core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -231,30 +293,14 @@ trait GenericSchema[R] extends DerivationSchema[R] with TemporalSchema {
     lazy val typeAName: String = Types.name(evA.toType_())
     lazy val typeBName: String = Types.name(evB.toType_())
 
-    objectSchema(
+    obj[RA with RB, (A, B)](
       s"Tuple${typeAName}And$typeBName",
-      Some(s"A tuple of $typeAName and $typeBName"),
-      (isInput, isSubscription) =>
-        List(
-          __Field(
-            "_1",
-            Some("First element of the tuple"),
-            Nil,
-            () =>
-              if (evA.optional) evA.toType_(isInput, isSubscription)
-              else makeNonNull(evA.toType_(isInput, isSubscription))
-          ) ->
-            ((tuple: (A, B)) => evA.resolve(tuple._1)),
-          __Field(
-            "_2",
-            Some("Second element of the tuple"),
-            Nil,
-            () =>
-              if (evB.optional) evB.toType_(isInput, isSubscription)
-              else makeNonNull(evB.toType_(isInput, isSubscription))
-          ) ->
-            ((tuple: (A, B)) => evB.resolve(tuple._2))
-        )
+      Some(s"A tuple of $typeAName and $typeBName")
+    )(implicit ft =>
+      NonEmptyChunk(
+        field[(A, B)]("_1", Some("First element of the tuple"))(_._1),
+        field[(A, B)]("_2", Some("Second element of the tuple"))(_._2)
+      )
```

Comment:
Could the helper be used for `lazy val kvSchema: Schema[RA with RB, (A, B)] = objectSchema(` a few lines below?

Reply:
Done